File: backend/schemas/photocard.py
from pydantic import BaseModel, Field
from typing import Optional, List

class PhotoData(BaseModel):
    id: str
    src: str
    objectPosition: Optional[str] = "center"
    objectFit: Optional[str] = "cover"
    scale: Optional[float] = 1.0

class StyleOverrides(BaseModel):
    backgroundColor: Optional[str] = None
    accentColor: Optional[str] = None
    headlineColor: Optional[str] = None

class PhotocardCreate(BaseModel):
    template_id: str
    headline: str
    subheadline: Optional[str] = ""
    brand_name: Optional[str] = "News Cards Studio"
    photos: List[PhotoData]
    style_overrides: Optional[StyleOverrides] = None
    # NEW: Capture the visual data from the frontend
    base64_image: Optional[str] = None 

class PhotocardResponse(BaseModel):
    id: str
    image_url: str
    status: str
    filename: str

File: frontend/src/types/photocard.ts
export interface PhotoSlot {
  id: string
  src: string | null
  objectPosition?: string
  objectFit?: string
  scale?: number
}

export interface TemplateStyle {
  accentColor?: string
  backgroundColor?: string
  headlineColor?: string
  subheadlineColor?: string
  brandBarBg?: string
  brandColor?: string
  headlineFontSize?: number
  subheadlineFontSize?: number
  headlineFontWeight?: number
  fontFamily?: string
  photoDividerColor?: string
  accentBarHeight?: number
  accentBarPosition?: 'top' | 'bottom'
  overlayOpacity?: number
  borderRadius?: number
  padding?: number
  showBrandBar?: boolean
  showAdBar?: boolean
  adBarBg?: string
  showWatermark?: boolean
  watermarkOpacity?: number
}

export interface PhotocardCreate {
  template_id: string
  headline: string
  subheadline?: string
  brand_name?: string
  photos?: PhotoSlot[]
  style_overrides?: Partial<TemplateStyle>
  base64_image?: string
  variant?: string
}

export interface PhotocardResponse {
  id: string
  image_url: string
  status: string
  filename: string
}
